pkg/actuators/machine/reconciler.go

```diff
@@ -342,6 +351,67 @@ func (r *Reconciler) updateLoadBalancers(instance *ec2.Instance) error {
 	return nil
 }
 
+// deletePlacementGroup deletes the placement group for the machine
+func (r *Reconciler) deletePlacementGroup() error {
+	placementGroupName := r.providerSpec.Placement.GroupName
+	if placementGroupName == "" {
+		// Nothing to do if the group name is empty
+		return nil
+	}
+
+	// Check that the placement group contains no instances
+	result, err := r.awsClient.DescribeInstances(&ec2.DescribeInstancesInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("placement-group-name"),
+				Values: []*string{aws.String(placementGroupName)},
+			},
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("could not get the number of instances in the placement group %s: %v", placementGroupName, err)
+	}
+
+	var instanceCount int
+	for _, reservation := range result.Reservations {
+		for range reservation.Instances {
+			instanceCount++
+		}
```

Comment:
I think this can be optimised
```suggestion
		instanceCount += len(reservation.Instances)
```